mqtt_as/mqtt_as.py

```diff
@@ -318,7 +318,8 @@ async def broker_up(self):  # Test broker connectivity
     async def disconnect(self):
         try:
             async with self.lock:
-                self._sock.write(b"\xe0\0")
+                if self._sock is not None:
```

Comment:
Maybe the following is neater - no need to grab the lock if already disconnected
```python
    async def disconnect(self):
        if self._sock is not None:
            try:
                async with self.lock:
                    self._sock.write(b"\xe0\0")
            except OSError:
                pass
            self._sock.close()
        self._has_connected = False
```

Reply:
Yes, that is better. I'll make that change.